src/broker/content-cache.c

```diff
@@ -858,12 +860,33 @@ static struct flux_msg_handler_spec handlers[] = {
     FLUX_MSGHANDLER_TABLE_END,
 };
 
+static flux_msg_handler_t *add_handler (flux_t *h, char *topic, uint32_t roles,
+                                        flux_msg_handler_f cb, void *arg)
+{
+    flux_msg_handler_t *w;
+    struct flux_match match = FLUX_MATCH_REQUEST;
+
+    match.topic_glob = topic;
+    if (!(w = flux_msg_handler_create (h, match, cb, arg)))
+        return NULL;
+    flux_msg_handler_allow_rolemask (w, roles);
+    flux_msg_handler_start (w);
+    return w;
+}
+
+
 int content_cache_set_flux (content_cache_t *cache, flux_t *h)
 {
     cache->h = h;
 
     if (flux_msg_handler_addvec (h, handlers, cache) < 0)
-	return -1;
+        return -1;
+    if (!(cache->load_w = add_handler (h, "content.load", FLUX_ROLE_USER,
+                                       content_load_request, cache)))
+        return -1;
+    if (!(cache->store_w = add_handler (h, "content.store", FLUX_ROLE_USER,
+                                        content_store_request, cache)))
+        return -1;
     if (flux_get_rank (h, &cache->rank) < 0)
```

Comment:
Ah that was actually an oversight - I meant to remove them from the handlers vector.

Agreed the interface needs work!

Can we just add a field to the end of the vector struct and assume it will be zeroed if not explicitly initialized?   I wasn't sure if it's wise to depend on it being zeroed.